File: FrontChatSystem.Api/FrontChatSystem.Api/Models/MessageWithReply.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Threading.Tasks;

namespace FrontChatSystem.Api.Models
{
    public class MessageWithReply
    {
        public HostMessage Message { get; set; }
        public IEnumerable<ReplyMessageData> ReplyMessage { get; set; }
    }

    public class HostMessage
    {
        public DateTime CreatedDateTime { get; set; }
        public string HostUserId { get; set; }
    }

    public class ReplyMessageData
    {
        public DateTime CreatedDateTime { get; set; }
        public string Content { get; set; }
        public string UserId { get; set; }
    }
}

File: FrontChatSystem/projects/front-chat-lib/src/lib/components/chat-window/chat/models/message-with-reply.ts
export class MessageWithReply implements IMessageWithReply {
    message: IHostMessage;
    replyMessage: IReplyMessageData[] | null;

    constructor(
        data: IMessageWithReply
    ) {
        this.message = new HostMessage(data.message);
        if (data.replyMessage) {
            this.replyMessage = data.replyMessage.map(m => {
                return new ReplyMessageData(m);
            });
        } else {
            this.replyMessage = null;
        }
    }
}

export interface IMessageWithReply {
    message: IHostMessage;
    replyMessage: IReplyMessageData[] | null;
}

export class HostMessage implements IHostMessage {
    createdDateTime: Date;
    hostUserId: string;

    constructor(data: IHostMessage) {
        this.createdDateTime = new Date(data.createdDateTime);
        this.hostUserId = data.hostUserId;
    }

}

export interface IHostMessage {
    createdDateTime: Date;
    hostUserId: string;
}

export class ReplyMessageData implements IReplyMessageData {
    createdDateTime: Date;
    content: string;
    userId: string;

    constructor(data: IReplyMessageData) {
        this.createdDateTime = new Date(data.createdDateTime);
        this.content = data.content;
        this.userId = data.userId;
    }
}


export interface IReplyMessageData {
    createdDateTime: Date;
    content: string;
    userId: string;
}
